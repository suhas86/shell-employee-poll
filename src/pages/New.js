import { useState } from 'react';
import { connect } from 'react-redux';
import { handleAddQuestion } from '../actions/questions';

const New = ({ dispatch }) => {
  const [optionOne, setOptionOne] = useState('');
  const [optionTwo, setOptionTwo] = useState('');

  const handleOnChange = (e) => {
    const { name, value } = e.target;
    if (name === 'optionOne') {
      setOptionOne(value);
    } else if (name === 'optionTwo') {
      setOptionTwo(value);
    }
  };
  const onSubmit = (e) => {
    e.preventDefault();
    dispatch(handleAddQuestion(optionOne, optionTwo));
    setOptionOne('');
    setOptionTwo('');
  };
  return (
    <div className="container">
      <h1>Would you rather</h1>
      <form onSubmit={onSubmit}>
        <label htmlFor="optionOne">
          First option
          <input
            id="optionOne"
            name="optionOne"
            data-testid="optionOne"
            type="text"
            value={optionOne}
            onChange={handleOnChange}
          />
        </label>
        <label htmlFor="optionTwo">
          Second option
          <input
            id="optionTwo"
            name="optionTwo"
            data-testid="optionTwo"
            type="text"
            value={optionTwo}
            onChange={handleOnChange}
          />
        </label>
        <button data-testid="submit" disabled={!optionOne || !optionTwo}>
          Submit
        </button>
      </form>
    </div>
  );
};

export default connect()(New);
