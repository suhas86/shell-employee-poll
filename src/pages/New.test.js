import { fireEvent, screen } from '@testing-library/react';
import renderWithRedux from '../utils/testing/renderWithRedux';
import { store } from '../index';
import New from './New';

describe('New Page', () => {
  it('renders without crashing', () => {
    const view = renderWithRedux(<New />, { store });

    expect(view).toMatchSnapshot();
  });
  it('Should not allow to post question if any one option is empty', () => {
    renderWithRedux(<New />, { store });
    const optionOne = screen.getByTestId('optionOne');
    fireEvent.change(optionOne, { target: { value: 'optionOne' } });
    const submitButton = screen.getByTestId('submit');
    expect(submitButton).toBeDisabled();
  });

  it('Should allow to post question if both options are filled', () => {
    renderWithRedux(<New />, { store });
    const optionOne = screen.getByTestId('optionOne');
    fireEvent.change(optionOne, { target: { value: 'optionOne' } });
    const optionTwo = screen.getByTestId('optionTwo');
    fireEvent.change(optionTwo, { target: { value: 'optionTwo' } });
    const submitButton = screen.getByTestId('submit');
    expect(submitButton).not.toBeDisabled();
  });
});
