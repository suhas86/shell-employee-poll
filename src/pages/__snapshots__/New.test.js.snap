// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Page renders without crashing 1`] = `undefined`;
